Use order tokens in payment capture

yariksheptykin opened the pull request:
Closes #2479

yariksheptykin committed:
Add a test

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/coreshop/CoreShop?pullRequest=2511) <br/>All committers have signed the CLA.

yariksheptykin committed:
Add route for paying with order token

yariksheptykin committed:
Change coreshop_payment definition

yariksheptykin commented on features/ui/frontend/checkout/prevent_repaying_completed_checkouts.feature:
```diff
@@ -0,0 +1,33 @@
+    And I submit the order
+    And I should be on the thank you page
+
+  #
```
```suggestion
```

yariksheptykin committed:
Remove empty comment

yariksheptykin committed:
New route for payments with tokens

yariksheptykin committed:
Migrate static routes

yariksheptykin commented:
@dpfaffenbauer I introduced a new route as you suggested and added a migration to update existing installations. Is there anything else we need to do to support BC?

dpfaffenbauer commented:
@yariksheptykin Amazing, I'll give it a test and let you know. Looks good though :)

yariksheptykin commented:
@dpfaffenbauer this does work at least on my local demo. Pay URL works both for token and for order id.

[Screencast from 19.01.2024 10:04:50.webm](https://github.com/coreshop/CoreShop/assets/636846/5c471409-267a-4ff2-9eb6-9dccf177c52f)

dpfaffenbauer commented:
@yariksheptykin since you backported it, we will close this one and merge the one for 3.2, I will merge 3.2 into 4.0 then.

Lovely screencast :), I will still have to test it on my own. Will happen sometime next week

yariksheptykin commented:
@dpfaffenbauer sounds good to me. I will patch our project (at neusta) with the backport and see it everything works in that context.

dpfaffenbauer commented:
https://github.com/coreshop/CoreShop/pull/2515

yariksheptykin commented:
@dpfaffenbauer https://github.com/coreshop/CoreShop/pull/2515 did not make it into v4.0.2. Is this intentional or just forgotten? Should I do anything to port the MR to v4?

dpfaffenbauer commented:
@yariksheptykin are you sure? 

https://github.com/coreshop/CoreShop/blob/4.0.2/src/CoreShop/Bundle/FrontendBundle/Resources/install/pimcore/staticroutes.yml#L145
https://github.com/coreshop/CoreShop/commit/c03814b7095ea2f531ca1792c06c5647b07d4b74

yariksheptykin commented:
@dpfaffenbauer ah I see you are right. I did not see this in release https://github.com/coreshop/CoreShop/releases/tag/4.0.2 and started worrying. Thanks for checking!

dpfaffenbauer commented:
@yariksheptykin It is a bit confusing right. CoreShop 4.0 always gets the changes from 3.2. 3.2 changes are in that changelog, only what changed from 3.2.x to 4.0.x are in the 4.0.x changelog.
